src/conversations/Conversations.ts

```diff
@@ -61,33 +67,46 @@ export class ConversationCache {
  */
 export default class Conversations {
   private client: Client
+  private v1Cache: ConversationCache
   private v2Cache: ConversationCache
 
   constructor(client: Client) {
     this.client = client
+    this.v1Cache = new ConversationCache()
     this.v2Cache = new ConversationCache()
   }
 
   /**
    * List all conversations with the current wallet found in the network, deduped by peer address
    */
   async list(): Promise<Conversation[]> {
-    const [seenPeers, v2Convos] = await Promise.all([
-      this.getIntroductionPeers(),
+    const [v1Convos, v2Convos] = await Promise.all([
+      this.listV1Conversations(),
       this.listV2Conversations(),
     ])
 
-    const v1Convos: Conversation[] = []
-    seenPeers.forEach((sent, peerAddress) =>
-      v1Convos.push(new ConversationV1(this.client, peerAddress, sent))
-    )
-
     const conversations = v1Convos.concat(v2Convos)
 
     conversations.sort((a, b) => a.createdAt.getTime() - b.createdAt.getTime())
     return conversations
   }
 
+  private async listV1Conversations(): Promise<Conversation[]> {
+    return this.v1Cache.load(async ({ latestSeen }) => {
```

Comment:
Neat change! Is it correct to say these v1 convos were not cached previously but are cached with this change?

Reply:
That's right